[client/catapult] Build Release error using GCC 12 and 13

Building Relase from source branch dev using GCC 12.3 and 13.4 on Ubuntu 24.04.
Dependencies being built using Conan package manager, hence "officially supported" build.

This error pops
```
In file included from /usr/include/string.h:535,
                 from /usr/include/c++/12/cstring:42,
                 from /home/andrea/.conan2/p/b/boost553a1f28e346f/p/include/boost/type_index/type_index_facade.hpp:18,
                 from /home/andrea/.conan2/p/b/boost553a1f28e346f/p/include/boost/type_index/stl_type_index.hpp:26,
                 from /home/andrea/.conan2/p/b/boost553a1f28e346f/p/include/boost/type_index.hpp:29,
                 from /home/andrea/.conan2/p/b/boost553a1f28e346f/p/include/boost/log/attributes/attribute_value_impl.hpp:18,
                 from /home/andrea/.conan2/p/b/boost553a1f28e346f/p/include/boost/log/attributes/constant.hpp:26,
                 from /home/andrea/symbol/client/catapult/src/catapult/utils/Logging.h:24,
                 from /home/andrea/symbol/client/catapult/src/catapult/exceptions.h:26,
                 from /home/andrea/symbol/client/catapult/src/catapult/utils/MemoryUtils.h:23,
                 from /home/andrea/symbol/client/catapult/src/catapult/ionet/Packet.h:24,
                 from /home/andrea/symbol/client/catapult/src/catapult/ionet/PacketEntityUtils.h:23,
                 from /home/andrea/symbol/client/catapult/tests/catapult/ionet/PacketEntityUtilsTests.cpp:22:
In function ‘void* memcpy(void*, const void*, size_t)’,
    inlined from ‘void catapult::utils::memcpy_cond(void*, const void*, size_t)’ at /home/andrea/symbol/client/catapult/src/catapult/utils/MemoryUtils.h:57:15,
    inlined from ‘static catapult::model::EntityRangeFactoryMixin<TEntity>::Range catapult::model::EntityRangeFactoryMixin<TEntity>::CopyFixed(const uint8_t*, size_t) [with TEntity = catapult::utils::ByteArray<catapult::Hash256_tag>]’ at /home/andrea/symbol/client/catapult/src/catapult/model/EntityRange.h:400:22,
    inlined from ‘catapult::model::EntityRange<TEntity> catapult::ionet::ExtractFixedSizeStructuresFromPacket(const Packet&) [with TStructure = catapult::utils::ByteArray<catapult::Hash256_tag>]’ at /home/andrea/symbol/client/catapult/src/catapult/ionet/PacketEntityUtils.h:74:77,
    inlined from ‘static auto catapult::ionet::{anonymous}::ExtractFixedSizeStructuresTraits::Extract(const catapult::ionet::Packet&)’ at /home/andrea/symbol/client/catapult/tests/catapult/ionet/PacketEntityUtilsTests.cpp:155:64,
    inlined from ‘void catapult::ionet::{anonymous}::AssertCannotExtractFromPacketHeaderWithNoData(uint32_t) [with TTraits = ExtractFixedSizeStructuresTraits]’ at /home/andrea/symbol/client/catapult/tests/catapult/ionet/PacketEntityUtilsTests.cpp:178:9:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:29:33: error: ‘void* __builtin_memcpy(void*, const void*, long unsigned int)’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
   29 |   return __builtin___memcpy_chk (__dest, __src, __len,
      |          ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
   30 |                                  __glibc_objsize0 (__dest));
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/andrea/symbol/client/catapult/tests/catapult/ionet/PacketEntityUtilsTests.cpp: In function ‘void catapult::ionet::{anonymous}::AssertCannotExtractFromPacketHeaderWithNoData(uint32_t) [with TTraits = ExtractFixedSizeStructuresTraits]’:
/home/andrea/symbol/client/catapult/tests/catapult/ionet/PacketEntityUtilsTests.cpp:173:32: note: at offset 8 into source object ‘packet’ of size 8
  173 |                         Packet packet;
      |                                ^~~~~~
cc1plus: all warnings being treated as errors
```

Definetly related to [this](https://github.com/symbol/symbol/blob/5756c8b8dfcb43e7ff1517df6271a115c8d76e1a/client/catapult/tests/catapult/ionet/PacketEntityUtilsTests.cpp#L165) but the older version is not considered.

--- a/client/catapult/tests/catapult/ionet/PacketEntityUtilsTests.cpp
+++ b/client/catapult/tests/catapult/ionet/PacketEntityUtilsTests.cpp
@@ -160,9 +160,9 @@ namespace catapult { namespace ionet {
 			}
 		};
 
-// disable this warning which happens only on GCC 14
+// disable this warning which happens since GCC 12
 // error: reading 1 or more bytes from a Region of size 0 [-Werror=stringop-overread]
-#if 14 <= __GNUC__ && !defined(__clang__)
+#if defined(__GNUC__) && !defined(__clang__) && (__GNUC__ >= 12)
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wstringop-overread"
 #endif
@@ -181,7 +181,7 @@ namespace catapult { namespace ionet {
 			EXPECT_TRUE(TTraits::IsEmpty(extractResult)) << "packet size " << size;
 		}
 
-#if 14 <= __GNUC__ && !defined(__clang__)
+#if defined(__GNUC__) && !defined(__clang__) && (__GNUC__ >= 12)
 #pragma GCC diagnostic pop
 #endif
 

